src/compute/src/render/mod.rs

```diff
@@ -343,466 +365,504 @@ pub fn build_compute_dataflow<A: Allocate>(
                 // Build declared objects.
                 for object in dataflow.objects_to_build {
                     let object_token = Rc::new(());
-                    context.shutdown_token = ShutdownToken::new(Rc::downgrade(&object_token));
+                    context.set_shutdown_token(ShutdownToken::new(Rc::downgrade(&object_token)));
                     tokens.insert(object.id, object_token);
 
-                    context.build_object(object);
+                    build_object(&mut context, object);
                 }
 
                 // Export declared indexes.
                 for (idx_id, dependencies, idx) in indexes {
-                    context.export_index(compute_state, &tokens, dependencies, idx_id, &idx);
+                    export_index(&context, compute_state, &tokens, dependencies, idx_id, &idx);
                 }
 
                 // Export declared sinks.
                 for (sink_id, dependencies, sink) in sinks {
-                    context.export_sink(compute_state, &tokens, dependencies, sink_id, &sink);
+                    export_sink(
+                        &context,
+                        compute_state,
+                        &tokens,
+                        dependencies,
+                        sink_id,
+                        &sink,
+                    );
                 }
             });
         }
     })
 }
 
-// This implementation block allows child timestamps to vary from parent timestamps,
-// but requires the parent timestamp to be `repr::Timestamp`.
```

Comment:
This comment still seems to apply?

Reply:
Well, there are no implementation blocks anymore, so I removed the "this implementation block" comments. I think this information is pretty clear from looking at the type annotations, but if you disagree we can add equivalent comments on the individual methods instead.